src/com/softwaret/kpdf/controller/login/LoginControllerImpl.kt

```diff
@@ -7,32 +7,41 @@ import com.softwaret.kpdf.response.Response
 import com.softwaret.kpdf.response.Unauthorized
 import com.softwaret.kpdf.response.error.ErrorResponseBody
 import com.softwaret.kpdf.response.success.LoginResponseBody
+import com.softwaret.kpdf.service.validation.input.InputValidator
+import com.softwaret.kpdf.util.extension.areAllNull
+import com.softwaret.kpdf.util.extension.get
 import com.softwaret.kpdf.util.parameters.BodyParameter.LOGIN
 import com.softwaret.kpdf.util.parameters.BodyParameter.PASSWORD
-import com.softwaret.kpdf.util.extension.get
-import com.softwaret.kpdf.util.extension.isAnyNotNull
 import io.ktor.http.Parameters
 
 class LoginControllerImpl(
-    private val loginInteractor: LoginInteractor
+    private val interactor: LoginInteractor,
+    private val inputValidator: InputValidator
 ) : BaseController(), LoginController {
 
     override suspend fun login(parameters: Parameters) =
         parameters[LOGIN, PASSWORD].let { (login, password) ->
-
             when {
-                isUserDataInvalid(login, password) -> Response.Unauthorized(ErrorResponseBody.AuthorizationFailed)
-                areCredentialsValid(login, password) -> Response.OK(LoginResponseBody("TOKEN"))
+                isInputAndUserDataValid(login, password).not() ->
+                    Response.Unauthorized(ErrorResponseBody.AuthorizationFailed)
+
+                areCredentialsValid(login, password) ->
+                    Response.OK(LoginResponseBody("TOKEN"))
+
                 else -> Response.Unauthorized(ErrorResponseBody.Unknown)
             }
         }
 
     private fun areCredentialsValid(login: String?, password: String?) =
-        loginInteractor.areCredentialsValid(login!!, password!!)
+        interactor.areCredentialsValid(login!!, password!!)
+
+    private fun isInputAndUserDataValid(login: String?, password: String?) =
+        isInputValid(login, password) && isUserDataValid(login!!, password!!)
```

Comment:
Funkcja `isInputValid` nie mówi względem czego waliduje. Mianowicie chodzi o fakt, że w funkcji `isUserDataValid` zakładamy nie-nullowalność parametrów, które wcześniej traktowaliśmy jako nullowalne, ja wiem dlaczego tak jest i Ty też, ale ktoś kto by to czytał musiałby by się naszukać odpowiedzi.